pallets/staking/src/lib.rs

```diff
@@ -3167,6 +3203,34 @@ impl<T: Trait> Module<T> {
             );
             let rest = max_payout.saturating_sub(validator_payout);
 
+            // Schedule Rewards for the validators
+            let next_block_no = <frame_system::Module<T>>::block_number() + 1.into();
+            for (index, validator_id) in T::SessionInterface::validators().into_iter().enumerate() {
```

Comment:
I think `end_era` is called after `new_era`. 
`new_era` updates the set of validators. Therefore, `T::SessionInterface::validators()` is likely returning the validators of next era instead of current/last era. We need to move this calculation to `new_era`, before we update the validator set.

Reply:
I don't think so, `end_era()` should be called before the `new_era()` because end_era() is initiated from the `end_session()` and end_session() will always be called first because it is hooked from on_initialize() variant of the session pallet. https://github.com/paritytech/substrate/blob/ba8e8122ab86bb1a8677b004ff4f14626fcc7884/frame/session/src/lib.rs#L551